src/main/java/com/google/devtools/build/lib/analysis/platform/PlatformUtils.java

```diff
@@ -84,38 +83,44 @@ public static Platform getPlatformProto(
       return null;
     }
 
-    Map<String, String> properties;
+    Map<String, String> properties = new HashMap<>();
     if (!spawn.getCombinedExecProperties().isEmpty()) {
       // Apply default exec properties if the execution platform does not already set
       // exec_properties
       if (spawn.getExecutionPlatform() == null
           || spawn.getExecutionPlatform().execProperties().isEmpty()) {
-        properties = new HashMap<>();
         properties.putAll(defaultExecProperties);
         properties.putAll(spawn.getCombinedExecProperties());
       } else {
         properties = spawn.getCombinedExecProperties();
       }
-    } else if (spawn.getExecutionPlatform() != null
-        && !Strings.isNullOrEmpty(spawn.getExecutionPlatform().remoteExecutionProperties())) {
+    } else if (spawn.getExecutionPlatform() != null) {
       properties = new HashMap<>();
-      // Try and get the platform info from the execution properties.
-      try {
+      String remoteExecutionProperties = spawn.getExecutionPlatform().remoteExecutionProperties();
+      if (!remoteExecutionProperties.isEmpty()) {
         Platform.Builder platformBuilder = Platform.newBuilder();
-        TextFormat.getParser()
-            .merge(spawn.getExecutionPlatform().remoteExecutionProperties(), platformBuilder);
+        // Try and get the platform info from the execution properties.
+        try {
+          TextFormat.getParser()
+                  .merge(spawn.getExecutionPlatform().remoteExecutionProperties(), platformBuilder);
+        } catch (ParseException e) {
+          String message =
+                  String.format(
+                          "Failed to parse remote_execution_properties from platform %s",
+                          spawn.getExecutionPlatform().label());
+          throw new UserExecException(
+                  e, createFailureDetail(message, Code.INVALID_REMOTE_EXECUTION_PROPERTIES));
+        }
+
         for (Property property : platformBuilder.getPropertiesList()) {
           properties.put(property.getName(), property.getValue());
         }
-      } catch (ParseException e) {
-        String message =
-            String.format(
-                "Failed to parse remote_execution_properties from platform %s",
-                spawn.getExecutionPlatform().label());
-        throw new UserExecException(
-            e, createFailureDetail(message, Code.INVALID_REMOTE_EXECUTION_PROPERTIES));
+      } else if (!spawn.getExecutionPlatform().execProperties().isEmpty()) {
```

Comment:
Why have the `isEmpty` check here? Without it, with an empty `execProperties`, the next statement is just `properties.putAll(empty map)`, which is effectively the same thing (I think).

Reply:
Half finished code refactor as well. This was before I moved the `defaultExecProperties` application on its own. Cleaned up in 6caeee6b97.